src/types/models/CommunityModel.jl

```diff
@@ -0,0 +1,228 @@
+
+    env_lbs = [envlink.lower_bound for envlink in cm.environmental_links]
+    env_ubs = [envlink.upper_bound for envlink in cm.environmental_links]
+
+    return ([models_lbs; env_lbs], [models_ubs; env_ubs])
+end
+
+Accessors.objective(cm::CommunityModel) = spzeros(n_variables(cm))
+
+function Accessors.coupling(cm::CommunityModel)
+    coups = blockdiag([coupling(m.model) for m in cm.members]...)
+    n = n_variables(cm)
+    return [coups spzeros(size(coups, 1), n - size(coups, 2))]
+end
+
+Accessors.n_coupling_constraints(cm::CommunityModel) =
+    sum(n_coupling_constraints(m.model) for m in cm.members)
+
+function Accessors.coupling_bounds(cm::CommunityModel)
+    lbs = vcat([first(coupling_bounds(m.model)) for m in cm.members]...)
+    ubs = vcat([last(coupling_bounds(m.model)) for m in cm.members]...)
+    return (lbs, ubs)
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Returns a matrix, which when multipled by the solution of a constraints based
+problem, yields the semantically meaningful fluxes that correspond to
+[`reactions`](@ref).
+"""
+function Accessors.reaction_variables_matrix(cm::CommunityModel)
+    rfs = blockdiag([reaction_variables_matrix(m.model) for m in cm.members]...)
+    nr = length(cm.environmental_links)
+    blockdiag(rfs, spdiagm(fill(1, nr)))
+end
+
+Accessors.reactions(cm::CommunityModel) = [
+    vcat([add_community_prefix.(Ref(m), reactions(m.model)) for m in cm.members]...)
+    [envlink.reaction_id for envlink in cm.environmental_links]
+]
+
+Accessors.n_reactions(cm::CommunityModel) =
+    sum(n_reactions(m.model) for m in cm.members) + length(cm.environmental_links)
+
+"""
+$(TYPEDSIGNATURES)
+
+Environmental reaction mapping to model variables.
+"""
+Accessors.environmental_reaction_variables(model::CommunityModel) = Dict(
+    rid => Dict(rid => 1.0) for
+    rid in [envlink.reaction_id for envlink in model.environmental_links]
+)
+
+#=
+This loops implements the rest of the accssors through access_community_member.
+Since most of the environmental reactions are generated programmtically, they
+will not have things like annotations etc. For this reason, these methods will
+only work if they access something inside the community members.
+=#
+for (func, def) in (
+    (:reaction_gene_associations, nothing),
+    (:reaction_subsystem, nothing),
+    (:reaction_stoichiometry, nothing),
+    (:metabolite_formula, nothing),
+    (:metabolite_charge, nothing),
+    (:metabolite_compartment, nothing),
+    (:reaction_annotations, Dict()),
+    (:metabolite_annotations, Dict()),
+    (:gene_annotations, Dict()),
+    (:reaction_notes, Dict()),
+    (:metabolite_notes, Dict()),
+    (:gene_notes, Dict()),
+    (:reaction_name, nothing),
+    (:metabolite_name, nothing),
+    (:gene_name, nothing),
+)
+    @eval begin
+        Accessors.$func(cm::CommunityModel, id::String) =
+            access_community_member(cm, id, $func; default = $def)
+    end
+end
```

Comment:
these should unmap the reaction ID to the community members and recurse to the contained models.

Reply:
still not sure what you mean here